test/modules/obsagent-client/src/config.rs

```diff
@@ -0,0 +1,173 @@
+use std::{num::ParseFloatError, string::ParseError};
+
+use clap::{value_t, App, Arg};
+use thiserror::Error;
+
+#[derive(Debug, Clone)]
+pub struct Config {
+    pub update_period: f64,
+    pub prom_config: PromConfig,
+    pub otel_config: OTelConfig,
+}
+
+#[derive(Debug, Clone)]
+pub struct OTelConfig {
+    pub push_period: f64,
+    pub otlp_endpoint: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct PromConfig {
+    pub endpoint: String,
+}
+
+#[derive(Debug, Error)]
+pub enum ConfigError {
+    #[error("error parsing command line argument into string: {0:?}")]
+    StringArgParseError(#[source] ParseError),
+    #[error("error parsing command line argument into float: {0:?}")]
+    FloatArgParseError(#[source] ParseFloatError),
+    #[error("invalid (negative) configuration parameter provided: {0:?}")]
+    InvalidNegativeArgVal(f64),
+    #[error("invalid (zero) configuration parameter provided: {0:?}")]
+    InvalidZeroArgVal(f64),
+}
+
+impl Config {
+    fn new(
+        update_period: f64,
+        prom_config: PromConfig,
+        otel_config: OTelConfig,
+    ) -> Result<Config, ConfigError> {
+        if update_period < 0.0 {
+            return Err(ConfigError::InvalidNegativeArgVal(update_period));
+        } else if update_period == 0.0 {
+            return Err(ConfigError::InvalidZeroArgVal(update_period));
+        }
+
+        Ok(Config {
+            update_period,
+            prom_config,
+            otel_config,
+        })
+    }
+}
+impl OTelConfig {
+    fn new(push_period: f64, otlp_endpoint: String) -> Result<OTelConfig, ConfigError> {
+        if push_period < 0.0 {
+            return Err(ConfigError::InvalidNegativeArgVal(push_period));
+        } else if push_period == 0.0 {
+            return Err(ConfigError::InvalidZeroArgVal(push_period));
+        }
+
+        Ok(OTelConfig {
+            push_period,
+            otlp_endpoint,
+        })
+    }
+}
+
+pub fn init_config() -> Result<Config, ConfigError> {
+    let matches = App::new("obs_agent_client")
+        .arg(
+            Arg::with_name("update-period")
+                .short("u")       
+                .long("update-period")
+                .takes_value(true)
+                .help("Period in seconds betweeen successive updates of each instrument with a new metric measurement.")
+        )
+        .arg(
+            Arg::with_name("push-period")
+                .short("p")
+                .long("push-period")
+                .takes_value(true)
+                .help("Period in seconds between successive pushes of  measurements  out of the client.")
```

Comment:
Nit: Could you mention the default value is 5s in the help log please?

Reply:
Done. Added `default_value` to clap `push_period` and `update_period` `Arg`s. When defaults are provided, clap conveniently updates the help output to show what the defaults are: